Public/Src/Utilities/Native/IO/Helpers.cs

```diff
@@ -63,6 +64,7 @@ public static class Helpers
             int postTimeoutMultiplier = DefaultPostTimeoutMultiplier)
         {
             Contract.Requires(work != null);
+            Contract.Requires(logExceptions || rethrowExceptions, "A caught exception must be either rethrown or logged.");

```

Comment:
>logExceptions || rethrowExceptions [](start = 30, length = 34)

`logExceptions ^ rethrowExceptions` ? Setting both to true does not log - it only throws.